generator/ServiceClientGeneratorLib/Generators/SourceFiles/StructureGenerator.tt

```diff
@@ -234,8 +234,20 @@ namespace <#=this.Config.Namespace#>.Model
 
                     // Because some services have model properties named "Equals" which conflicts with .NET's Equals operation. Add the "new" keyword to override the behavior.
                     // The Retryable is there because the CloudHsmServiceException exception from CloudHSM models Retryable with a different return then what comes from the base class.
-#>
-        <#=member.AccessModifier#> <#=member.PropertyName.Equals("Equals") || (this.Structure is ExceptionShape && member.PropertyName.Equals("Retryable")) ? "new " : ""#><#=member.DetermineType()#> <#=member.PropertyName#>
+                    var shouldAddNewKeyword = 
+                        member.PropertyName.Equals("Equals") || 
+                        (this.Structure is ExceptionShape && member.PropertyName.Equals("Retryable"));
+
+                    // Cognito and SecurityToken also have a modeled property named Expiration, which conflicts with the interface used in SRA.
```

Comment:
Both Cognito [1] and SecurityToken [2] have hand-written code that extend `AWSCredentials`. If I didn't add the `new` keyword, the solution would not compile.

[1] https://github.com/aws/aws-sdk-net/blob/main/sdk/src/Services/CognitoIdentity/Custom/Model/Credentials.Extensions.cs
[2] https://github.com/aws/aws-sdk-net/blob/main/sdk/src/Services/SecurityToken/Custom/Model/Credentials.Extensions.cs

Reply:
Since this is V4 I think we should just remove the `Expiration in the Cognito and SecurityToken credentials and let them just use the base class implementation.

I definitely don't want to use the `new` as that could cause problems of the wrong `Expiration` property being used depending on the cast of the credentials variable.

Reply:
I miss understood these were generated classes that we added a `partial` to make the be `AWSCredentials` objects.